### Bioinformatics for Next Generation Sequencing

Homeworks for the course.

List of used software:

* [Samtools](http://www.htslib.org/)
* [HISAT2](https://ccb.jhu.edu/software/hisat2/index.shtml)
* [StringTie](https://ccb.jhu.edu/software/stringtie/)
* [HTSeq](https://htseq.readthedocs.io/en/release_0.11.1/count.html)
* [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
* [Entrez Direct](https://www.ncbi.nlm.nih.gov/home/tools/)
* [SRA Toolkit](https://www.ncbi.nlm.nih.gov/home/tools/)
* [Cutadapt](https://cutadapt.readthedocs.io/en/stable/)
* [SeqKit](https://bioinf.shenwei.me/seqkit/)
* [Bowtie](http://bowtie-bio.sourceforge.net/index.shtml)
* [plastid](https://plastid.readthedocs.io/en/latest/index.html)
* [bedtools](https://bedtools.readthedocs.io/en/latest/)
* [svist4get](https://bitbucket.org/artegorov/svist4get/src/master/)
* [deepTools](https://deeptools.readthedocs.io/en/develop/)
* [3D Genome Browser](http://promoter.bx.psu.edu/hi-c/)
* [HiGlass](http://higlass.io/)
* [Cooler](https://cooler.readthedocs.io/en/latest/)
* [HiCExplorer](https://hicexplorer.readthedocs.io/en/latest/)
* [BWA](http://bio-bwa.sourceforge.net/)
* [Picard](https://broadinstitute.github.io/picard/)
* [IGV](https://software.broadinstitute.org/software/igv/)
* [GATK](https://software.broadinstitute.org/gatk/)
* [SnpEff](http://snpeff.sourceforge.net/)
* R packages:
   - [edgeR](https://bioconductor.org/packages/release/bioc/html/edgeR.html)
   - [Tidyverse](https://www.tidyverse.org/)
   - [Rsubread](https://bioconductor.org/packages/release/bioc/html/Rsubread.html)
   - [topGO](https://bioconductor.org/packages/release/bioc/html/topGO.html)
   - [Seurat](https://satijalab.org/seurat/)
   - [velocyto](http://velocyto.org/)
   - [Monocle 3](https://cole-trapnell-lab.github.io/monocle3/)